public/app/plugins/datasource/grafana-azure-monitor-datasource/components/ResourcePicker/NestedRows.tsx

```diff
@@ -1,66 +1,98 @@
 import { cx } from '@emotion/css';
 import { Checkbox, HorizontalGroup, Icon, IconButton, useStyles2, useTheme2 } from '@grafana/ui';
-import React, { Fragment, useCallback } from 'react';
+import React, { useCallback, useState } from 'react';
 import getStyles from './styles';
-import { EntryType, Row } from './types';
+import { EntryType, Row, RowGroup } from './types';
 
 interface NestedRowsProps {
-  rows: Row[];
+  rows: RowGroup;
   level: number;
-  selected: string[];
-  onRowToggleCollapse: (row: Row) => void;
+  selectedRows: RowGroup;
+  fetchNested: (row: Row) => Promise<void>;
   onRowSelectedChange: (row: Row, selected: boolean) => void;
 }
 
-const NestedRows: React.FC<NestedRowsProps> = ({ rows, selected, level, onRowToggleCollapse, onRowSelectedChange }) => {
+const NestedRows: React.FC<NestedRowsProps> = ({ rows, selectedRows, level, fetchNested, onRowSelectedChange }) => (
+  <>
+    {Object.keys(rows).map((rowId) => (
+      <NestedRow
+        key={rowId}
+        row={rows[rowId]}
+        selectedRows={selectedRows}
+        level={level}
+        fetchNested={fetchNested}
+        onRowSelectedChange={onRowSelectedChange}
+      />
+    ))}
+  </>
+);
+
+interface NestedRowProps {
+  row: Row;
+  level: number;
+  selectedRows: RowGroup;
+  fetchNested: (row: Row) => Promise<void>;
+  onRowSelectedChange: (row: Row, selected: boolean) => void;
+}
+
+const NestedRow: React.FC<NestedRowProps> = ({ row, selectedRows, level, fetchNested, onRowSelectedChange }) => {
   const styles = useStyles2(getStyles);
 
+  const isSelected = !!selectedRows[row.id];
+  const isDisabled = Object.keys(selectedRows).length > 0 && !isSelected;
+  const initialOpenStatus = row.type === EntryType.Collection ? 'open' : 'closed';
+  const [openStatus, setOpenStatus] = useState<'open' | 'closed' | 'loading'>(initialOpenStatus);
+  const isOpen = openStatus === 'open';
+
+  const onRowToggleCollapse = async () => {
+    if (openStatus === 'open') {
+      setOpenStatus('closed');
+      return;
+    }
+    setOpenStatus('loading');
+    await fetchNested(row);
```

Comment:
should we skip the fetching if the row has been fetched before? (not sure if the content may change over time)